lib/elixir/lib/map.ex

```diff
@@ -371,6 +371,26 @@ defmodule Map do
     :maps.update(key, value, map)
   end
 
+  @doc """
+  Replaces the value under `key` using the given function only if
+  `key` already exists in `map`.
+
+  If `key` does not exist, the original map is returned unchanged.
+
+  ## Examples
+      iex> Map.replace_lazy(%{a: 1, b: 2}, :a, fn v -> v * 4 end)
+      %{a: 4, b: 2}
+
+      iex> Map.replace_lazy(%{a: 1, b: 2}, :c, fn v -> v * 4 end)
+      %{a: 1, b: 2}
+  """
+  @doc since: "1.14.0"
+  @spec replace_lazy(map, key, (existing_value :: value -> new_value :: value)) :: map
+  def replace_lazy(map, key, _) when not is_map_key(map, key), do: map
+  def replace_lazy(map, key, fun) when is_function(fun, 1) do
+    Map.replace(map, key, fun.(map[key]))
```

Comment:
This is not lazy, `fun/1` will be evaluated even if the key does not exist on the map.

Reply:
If it doesn't exist in the map, the first declaration hits.

Reply:
Ah, of course you're right.